### PR TITLE
Add example gallery

### PR DESCRIPTION
**Pull request by erikmannerfelt**
The example gallery seems to be working!

Now, an example in the `examples/` directory will show up in the `Example gallery`, and will link to the API if a function from xdem is used. It will also link to the documentation of `geoutils`, `numpy` and `matplotlib` (more can be added).

There are only two weird things:

- Things defined in the `__init__` does not really work (at least now; I'm sure it's fixable). Therefore, `xdem.DEM` refers to `gu.Raster` (since it can't find the definition of `xdem.DEM`). EDIT: This is now fixed!
- Each example has to be prepended with `plot_` in order to actually execute. This is according to https://github.com/sphinx-gallery/sphinx-gallery, and might be possible to circumvent. Still, I think it's a nice way to know which functions will run and which will only render as notebook-like code.

Here's the documentation!
https://xdem-erikm.readthedocs.io/en/example-gallery/index.html

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -43,20 +46,35 @@
...
-autosummary_generate = True
+#autosummary_generate = True
+
+intersphinx_mapping = {
+    "geoutils": ("https://geoutils.readthedocs.io/en/latest", None),
+    "numpy": ("https://numpy.org/doc/stable", None),
+    "matplotlib": ("https://matplotlib.org/stable", None),
+}
+
+sphinx_gallery_conf = {
```
Comment (rhugonnet) [inline docs/source/conf.py:62]: reads like that wasn't easy :D

**Review by rhugonnet — APPROVED** [review]
Very nice :+1:

File: docs/source/conf.py (hunk 1 above)
Comment (erikmannerfelt) [reply docs/source/conf.py:62]: Nope hahaha!

**Comment by erikmannerfelt** [comment]
Thanks for the approval, @rhugonnet! I'll add some more examples today and tweak some settings, so I'll request a review again once that is ready.

**Comment by erikmannerfelt** [comment]
Now, the documentation is ready for review!

Two things that I've included that are only indirectly relevant to the documentation:
* The "longyearbyen_tba_dem_coreg" key now works for `xdem.examples.get_path`
* Some small `xdem.coreg.apply_matrix` tweaks were made to increase its fidelity.

**Review by rhugonnet — APPROVED** [review]
I don't have much to say except that it's great!
I'm thinking that maybe in the future we'll have to refine the scope of the examples depending on their relation to other examples, or possibly to try to be more concise to avoid discouraging new users.
But right now I think it's perfect :)

**Comment by erikmannerfelt** [comment]
Thank you, @rhugonnet! There are for example ways of categorizing examples into e.g. "Basic", "Intermediate" and "Advanced" categories.

File: docs/source/_templates/module.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{{ fullname }}
```
Comment (adehecq) [inline docs/source/_templates/module.rst:1]: It's all black magic to me! ;-)

File: docs/source/conf.py (hunk 1 above)
Comment (adehecq) [reply docs/source/conf.py:63]: So that's used to link to the documentation of these packages? Great!!

**Pull request merged**